src/components/kepler-gl.js

```diff
@@ -138,30 +139,40 @@ export const mapFieldsSelector = props => ({
   locale: props.uiState.locale
 });
 
-export const sidePanelSelector = (props, availableProviders) => ({
-  appName: props.appName,
-  version: props.version,
-  appWebsite: props.appWebsite,
-  mapStyle: props.mapStyle,
-  onSaveMap: props.onSaveMap,
-  uiState: props.uiState,
-  mapStyleActions: props.mapStyleActions,
-  visStateActions: props.visStateActions,
-  uiStateActions: props.uiStateActions,
+export function getVisibleDatasets(datasets) {
+  // We don't want Geocoder dataset to be present in SidePanel dataset list
+  return filterObjectByPredicate(datasets, (key, value) => key !== GEOCODER_DATASET_NAME);
+}
 
-  datasets: props.visState.datasets,
-  filters: props.visState.filters,
-  layers: props.visState.layers,
-  layerOrder: props.visState.layerOrder,
-  layerClasses: props.visState.layerClasses,
-  interactionConfig: props.visState.interactionConfig,
-  mapInfo: props.visState.mapInfo,
-  layerBlending: props.visState.layerBlending,
+export const sidePanelSelector = (props, availableProviders) => {
+  // visibleDatasets
+  const filteredDatasets = getVisibleDatasets(props.visState.datasets);
```

Comment:
The problem here is that the thing that should be memoized is the reference to props.visState.datasets, not its content.
KeplerGL's state and visState will change onHover, this means the render function will be called invoking the sidePanelSelector function.
This function will use props.visState.datasets that are passed to SidePanels.
The objective is to filter out the GEOCODER_DATASET and we can do that either in a mutable or immutable way.
If we do it in an immutable way then we lose the reference since we are creating a new object (current solution).
The only solution is to do it in a mutable way where we change the props.visState.datasets object but keep the reference.
The dirty way is to just do something like: delete props.visState.datasets[GEOCODER_DATASET_NAME] (which works) but something better should be implemented if possible (will try).

Reply:
you can create selector for `filteredDatasets`

```js
  // component method:
  filteredDatasetsSelector = createSelector(props => props.visState.datasets, getVisibleDatasets);

  // in render function
  const filteredDatasets = this.filteredDatasetsSelector(this.props);
  // add filteredDatasets as additional argument to sidePanelSelector
  const sideFields = sidePanelSelector(this.props, availableProviders, filteredDatasets);
```

Reply:
This doesn't solve the issue of calling render for LayerManager (or its children). Memoization needs to be done in children too.

Reply:
Shan and I agreed that we add the filteredDatasetsSelector in order to not change the datasets reference, but the other issues will be handled some other time.